test/e2e/placement_selecting_resources_test.go

```diff
@@ -592,3 +594,433 @@ var _ = Describe("validating CRP when deleting resources in a matching namespace
 		Eventually(finalizerRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to remove controller finalizers from CRP %s", crpName)
 	})
 })
+
+// TODO should be blocked by the validation webhook
```

Comment:
Just for clarification, does the validation webhook check for this?

Reply:
'cause for this specific test case, i.e., non-existent kind, it would require the webhook to know which APIs are registered. It also excludes the experience where a CRP is created before the API becomes available (a very rare case though).

Reply:
i'm thinking of blocking them at the very beginning as it's not that obvious for the customer to notice the failure during the runtime.

@ryanzhang-oss what do you think?

Reply:
IMO, the validation webhook, in general, should not go so deep into CRP controller's logic.  Currently, it checks if the resource selector is valid, but it doesn't check if the selector selects anything. The line is a bit grey though.  Currently, it doesn't check if the resource selector's GVK is valid for v1beta1. We check it for v1alpha1.

Reply:
However, the webhook wouldn't check if the selector select anything as resources can be applied later.

Reply:
There are two cases:
1. the GVK is invalid.
2. the GVK is valid while the resource does not exist.

Current behavior:
1. For the case1, CRP controller sets the scheduled condition as false to return the error.
2. For the case2, CRP controller won't return the error and just select nothing.

For the case2, webhook should not block it, either.
For the case1, @ryanzhang-oss are you suggesting keeping it consistent with v1alpha1 (block by webhook)?

btw, will add a test for case2.

Reply:
Yes, for case 1, v1beta1 webhook will try to block it (it doesn't have the absolute truth though).